modules/katello_composite_content_view.py

```diff
@@ -0,0 +1,176 @@
+    content_views:
+      - name: 'Fedora CV'
+        version: '2.0'
+      - name: 'Ansible CV'
+        version: '1.0'
+'''
+
+RETURN = '''# '''
+
+try:
+    from ansible.module_utils.ansible_nailgun_cement import (
+        create_server,
+        find_organization,
+        find_content_views,
+        find_content_view,
+        find_content_view_version,
+        ping_server,
+        set_task_timeout,
+    )
+    HAS_NAILGUN_PACKAGE = True
+except:
+    HAS_NAILGUN_PACKAGE = False
+from ansible.module_utils.foreman_helper import handle_no_nailgun
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.foreman_helper import handle_no_nailgun
+
+
+def content_view(module, name, organization, content_views=[]):
+    updated = False
+    organization = find_organization(module, organization)
+
+    content_view = find_content_view(module, name=name, organization=organization)
+    response = content_view.search()
+
+    if len(response) == 1:
+        content_view = response[0]
+    elif len(response) == 0:
+        if not module.check_mode:
+            content_view = content_view.create()
+        updated = True
+
+    content_views = find_content_views(content_views, organization)
+
+    if set(map(lambda r: r.id, content_views)) != set(map(lambda r: r.id, content_view.repository)):
+        if not module.check_mode:
+            content_view.repository = content_views
+            content_view.update(['repository'])  # TODO ?
+        updated = True
+
+    return updated
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            server_url=dict(required=True),
+            username=dict(required=True, no_log=True),
+            password=dict(required=True, no_log=True),
+            verify_ssl=dict(type='bool', default=True),
+            name=dict(required=True),
+            organization=dict(required=True),
+            content_views=dict(type='list'),
+        ),
+        supports_check_mode=True,
+    )
+
+    handle_no_nailgun(module, HAS_NAILGUN_PACKAGE)
+
+    server_url = module.params['server_url']
+    username = module.params['username']
+    password = module.params['password']
+    verify_ssl = module.params['verify_ssl']
+    name = module.params['name']
+    organization = module.params['organization']
+    content_views = module.params['content_views']
+
+    server = ServerConfig(
+        url=server_url,
+        auth=(username, password),
+        verify=verify_ssl
+    )
+    ng = NailGun(server, entities, module)
+
+    # Lets make an connection to the server with username and password
+    try:
+        org = entities.Organization(server)
+        org.search()
+    except Exception as e:
+        module.fail_json(msg="Failed to connect to Foreman server: %s " % e)
+
+    kwargs = {}
+    if content_views:
+        kwargs['content_views'] = content_views
+
+    try:
+        changed = ng.content_view(name, organization, **kwargs)
```

Comment:
Could we use `naildown_entity_state` with `sanitize_composite_content_view_dict`?

Reply:
whats `sanitize_composite_content_view_dict` ?

Reply:
so i found the reference for this... almost seems like there should be a `sanitize_dict` in module_utils hat takes a map from each object type...  all 9 current `sanitize_*_dict` are just separate functions with a different map and the same 5 lines of code finishing the function

Reply:
I agree this could be more DRY. If you'd like, we can open an issue to address it, and ignore the repetitiveness for this PR. I'm not sure Nailgun is the appropriate place for it.

Reply:
I have a branch to start cleaning this up, but i wanted to wait for the automated tests for this kind of refactoring.